DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/svd/src/post_process.hpp

```diff
@@ -0,0 +1,259 @@
+    constexpr int a_block_count = A_rows / A_cols;
+    // Copy a matrix from the pipe to a local memory
+    // Number of pipe reads of pipe_size required to read a full column
+    constexpr int rExtraIteration = ((A_cols % pipe_size) != 0) ? 1 : 0;
+    constexpr int sExtraIteration = ((A_rows % pipe_size) != 0) ? 1 : 0;
+    constexpr int aExtraIteration = ((A_cols % pipe_size) != 0) ? 1 : 0;
+    constexpr int vExtraIteration = ((A_cols % pipe_size) != 0) ? 1 : 0;
+    constexpr int uExtraIteration = ((A_rows % pipe_size) != 0) ? 1 : 0;
+    constexpr int rLoopIterPerColumn = (A_cols / pipe_size) + rExtraIteration;
+    constexpr int sLoopIterPerColumn = (A_rows / pipe_size) + sExtraIteration;
+    constexpr int aLoopIterPerColumn = (A_cols / pipe_size) + aExtraIteration;
+    constexpr int vLoopIterPerColumn = (A_cols / pipe_size) + vExtraIteration;
+    constexpr int uLoopIterPerColumn = (A_rows / pipe_size) + uExtraIteration;
+    // Number of pipe reads of pipe_size to read all the matrices
+    constexpr int rLoopIter = rLoopIterPerColumn * A_cols;
+    constexpr int sLoopIter = sLoopIterPerColumn * A_cols;
+    constexpr int aLoopIter = aLoopIterPerColumn * A_cols;
+    constexpr int vLoopIter = vLoopIterPerColumn * A_cols;
+    constexpr int uLoopIter = uLoopIterPerColumn * A_rows;
+    // Size in bits of the loop iterator over kLoopIter iterations
+    // constexpr int rLoopIterBitSize =
+    //     fpga_tools::BitsForMaxValue<rLoopIter + 1>();
+    constexpr int sLoopIterBitSize =
+        fpga_tools::BitsForMaxValue<sLoopIter + 1>();
+    constexpr int aLoopIterBitSize =
+        fpga_tools::BitsForMaxValue<aLoopIter + 1>();
+    constexpr int vLoopIterBitSize =
+        fpga_tools::BitsForMaxValue<rLoopIter + 1>();
+    constexpr int uLoopIterBitSize =
+        fpga_tools::BitsForMaxValue<uLoopIter + 1>();
+
+    constexpr short kBankwidth = pipe_size * sizeof(TT);
+    constexpr unsigned short sNumBanks = A_rows / pipe_size;
+    constexpr unsigned short aNumBanks = A_cols / pipe_size;
+    constexpr unsigned short vNumBanks = A_rows / pipe_size;
+    constexpr unsigned short uNumBanks = A_rows / pipe_size;
+    constexpr short sNumBanksNextPow2 =
+        fpga_tools::Pow2(fpga_tools::CeilLog2(sNumBanks));
+    constexpr short aNumBanksNextPow2 =
+        fpga_tools::Pow2(fpga_tools::CeilLog2(aNumBanks));
+    constexpr short vNumBanksNextPow2 =
+        fpga_tools::Pow2(fpga_tools::CeilLog2(vNumBanks));
+    constexpr short uNumBanksNextPow2 =
+        fpga_tools::Pow2(fpga_tools::CeilLog2(uNumBanks));
+
+    while (1) {
+      [[intel::numbanks(sNumBanksNextPow2)]]  // NO-FORMAT: Attribute
+      [[intel::bankwidth(kBankwidth)]]        // NO-FORMAT: Attribute
+      [[intel::private_copies(4)]]            // NO-FORMAT: Attribute
+      [[intel::max_replicates(1)]]            // NO-FORMAT: Attribute
+      TT S_result[A_rows][A_cols];
+
+      [[intel::numbanks(aNumBanksNextPow2)]]  // NO-FORMAT: Attribute
```

Comment:
I thought there was a bugfix for not requiring power-of-2 numbers of banks already. Please double check if this is still necessary.

Reply:
I am having trouble using just the `aNumBanks` and getting:
```
error: 'numbanks' attribute requires a positive integral compile time constant expression
```
Its odd since `aNumBnaks` is a `constexpr short`